core/src/main/java/ai/timefold/solver/core/impl/domain/entity/descriptor/EntityDescriptor.java

```diff
@@ -913,6 +901,74 @@ public boolean hasNoNullVariables(Object entity) {
         };
     }
 
+    private Predicate<Object> forEachIncludingPredicateBasicVarSupplier(EnumSet<ForEachInclude> includeSet) {
+        // TODO: Use a supply for this?
+        final var inconsistentDescriptor = this.getShadowVariablesInconsistentDescriptor();
+        final Predicate<Object> isUnassignedPredicate = this::hasNoNullVariables;
+        final Predicate<Object> isInconsistentPredicate = (inconsistentDescriptor == null) ? ignored -> true
+                : entity -> Boolean.FALSE.equals(inconsistentDescriptor.getValue(entity));
+
+        if (includeSet.isEmpty()) {
+            return entity -> isUnassignedPredicate.test(entity) && isInconsistentPredicate.test(entity);
+        }
+
+        var excludeToPredicate = Map.of(
```

Comment:
`EnumMap`? Doesn't matter much in this case though.

Reply:
`EnumMap` does not have an `of` API, which makes code using it harder to read/modify.